packages/@aws-cdk/aws-codepipeline-actions/lib/s3/deploy-action.ts

```diff
@@ -86,6 +86,14 @@ export interface S3DeployActionProps extends codepipeline.CommonAwsActionProps {
    * @default - none, decided by the HTTP client
    */
   readonly cacheControl?: CacheControl[];
+
+  /**
+   * The ARN of the AWS KMS encryption key for the host bucket.
+   * The KMSEncryptionKeyARN parameter encrypts uploaded artifacts with the provided AWS KMS key.
+   * For a KMS key, you can use the key ID, the key ARN, or the alias ARN.
+   * @default - none
+   */
+  readonly kmsEncryptionKeyArn?: string;
```

Comment:
Rather than this being an ARN, we should be taking in an `IKey` here.

Reply:
Hi Kendra, Thanks for reviewing this :)

I have updated the code.